dataset/filter.py

```diff
@@ -0,0 +1,370 @@
+    # Only filter out single alphabetic words (i.e. leave in e.g. Chinese)
+    if len(example["subject"]) < 4 or (len(example["subject"].split()) == 1 and example["subject"].isalpha()):
+        return False
+    return True
+
+ds = ds.filter(filter_empty_messages, num_proc=NUM_PROC)
+
+print("After empty message filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.map(clean_issues_and_refs, num_proc=NUM_PROC)
+
+ds = ds.filter(filter_empty_messages, num_proc=NUM_PROC)
+
+print("After empty message filtering due to messages with []:, the dataset size is: {}".format(len(ds)))
+
+# Old mechanism with only partial files
+"""
+ds = ds.map(get_line_diff_range, num_proc=NUM_PROC)
+
+def filter_length(example):
+    if example["old_change_range"] <= 2:
+        if random.random() > SHORT_SAMPLING:
+            return False
+
+    if example["old_change_range"] >= 200:
+        if random.random() > LONG_SAMPLING:
+            return False
+
+    return True
+
+ds = ds.filter(filter_length, num_proc=NUM_PROC)
+"""
+
+if MODEL == "santacoder":
+    from transformers import AutoTokenizer
+    tokenizer = AutoTokenizer.from_pretrained("bigcode/santacoder")
+    # Filter for texts with with less than 2048 tokens
+    ds = ds.filter(lambda x: len(tokenizer("<|endoftext|>" + x["old_contents"] + "<|endoftext|>" + x["subject"] + "<|endoftext|>" + x["new_contents"])["input_ids"]) <= 2048, num_proc=NUM_PROC)
+elif MODEL == "bigcode":
+    from transformers import AutoTokenizer
+    tokenizer = AutoTokenizer.from_pretrained("bigcode/large-model")
+    # Filter for texts with with less than 8192 tokens
+    ds = ds.filter(lambda x: len(tokenizer("<|endoftext|>" + x["old_contents"] + "<|endoftext|>" + x["subject"] + "<|endoftext|>" + x["new_contents"] + "<|endoftext|>")["input_ids"]) <= 8192, num_proc=NUM_PROC)
+elif MODEL == "bloomz":
+    from transformers import AutoTokenizer
+    tokenizer = AutoTokenizer.from_pretrained("bigscience/bloomz-7b1")
+    ds = ds.filter(lambda x: len(tokenizer(f"{x['old_contents']}\n\n{x['subject']}\n{x['new_contents']}")["input_ids"]) <= 2048, num_proc=NUM_PROC)
+elif MODEL == "codegeex":
+    from transformers.models.gpt2 import GPT2TokenizerFast
+    tokenizer = GPT2TokenizerFast.from_pretrained("gpt2")
+    # CodeGeeX has some extra tokenization to use less tokens for many whitespaces so be a bit less strict
+    ds = ds.filter(lambda x: len(tokenizer(f"{x['old_contents']}{x['subject']}{x['new_contents']}")["input_ids"]) <= 2048, num_proc=NUM_PROC)
+
+print("After length filtering, the dataset size is: {}".format(len(ds)))
+
+def filter_messages(example):
+    lower_subject = example["subject"].lower()
+
+    # remove samples without desired start words or with low proba
+    if not (lower_subject.startswith(tuple(GOOD_STARTS))) \
+    and not (lower_subject.endswith(tuple(GOOD_ENDS))) \
+    and (("proba" not in example) or (example["proba"] < 0.1)):
```

Comment:
We may be careful since there is no `proba` filed for the whole dataset 🤔

Reply:
Yeah hence the check if "proba" not in example; We can also safely remove it